src/StatusQ/Controls/StatusInput.qml

```diff
@@ -0,0 +1,85 @@
+import QtQuick 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+Rectangle {
+    id: root
+
+    implicitWidth: 480
+    height: (label.visible ? 
+                  label.anchors.topMargin +
+                  label.height :
+              charLimitLabel.visible ?
+                  charLimitLabel.anchors.topMargin +
+                  charLimitLabel.height :
+              0) +
+            statusBaseInput.anchors.topMargin +
+            statusBaseInput.height + 
+            (errorMessage.visible ? 
+                  errorMessage.anchors.topMargin +
+                  errorMessage.height :
+                  0) + 8
+
+    color: "transparent"
+    property alias input: statusBaseInput
+    property string label: ""
+    property int charLimit: 0
+    property string errorMessage: ""
+
+    StatusBaseText {
+        id: label
+        height: visible ? implicitHeight : 0
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.topMargin: visible ? 8 : 0
+        anchors.leftMargin: 16
+        visible: !!root.label
+
+        text: root.label
+        font.pixelSize: 15
+        color: statusBaseInput.enabled ? Theme.palette.directColor1 : Theme.palette.baseColor1
+    }
+
+    StatusBaseText {
+        id: charLimitLabel
+        height: visible ? implicitHeight : 0
+        anchors.top: parent.top
+        anchors.right: parent.right
+        anchors.topMargin: visible ? 11 : 0
+        anchors.rightMargin: 16
+        visible: !!root.charLimit
+
+        text: "0 / " + root.charLimit
```

Comment:
probably not implemented yet, because the characters counter is not updated.
`"%1 / %2".arg(statusBaseInput.text).arg(root.charLimit)`
also it looks like the charLimit is not enforced on `statusBaseInput`.

Reply:
> probably not implemented yet, because the characters counter is not updated.
"%1 / %2".arg(statusBaseInput.text.length).arg(root.charLimit)

Totally missed this. Thanks for the quick snippet

The enforcement should be done via a validator. What "enforcement" means though, depends on the application. I could imagine that enforcement of char limit isn't needed, but setting the validity of this input is, so that other controls can be disabled accordingly (like submit buttons).

Either way, validator pipeline still needs to be built.